modules/core/src/views/view-manager.js

```diff
@@ -161,9 +161,21 @@ export default class ViewManager {
       this._setSize(props.width, props.height);
     }
 
-    this._rebuildViewportsFromViews();
+    if (!this._isUpdating) {
+      this._update();
+    }
+  }
+
+  _update() {
+    this._isUpdating = true;
```

Comment:
I suspect the reason for this guarding logic will not be obvious to someone who hasn't just been debugging this issue. The short comment you left as a hint inside didn't quite suffice for me, better if it goes next to the if statements.

I think we'd benefit from a slightly more extensive comment here at the top. You could just paste your explanation from the PR header.